### PR TITLE
Allow value zero for TestSessionTimeOut 

### PR DESCRIPTION
**Pull request by Faizan2304**
Allow value zero for TestSessionTimeOut and in this case TestSessionTimeOut will be infinite. Or in other word we will not check for testSession timeout.

DEVDIV Bug: https://devdiv.visualstudio.com/DevDiv/VS.in%20Agile%20Testing%20IDE/_workitems/edit/497360

### CODE REVIEW

**Comment by Faizan2304** [comment]
@dotnet-bot test this Windows_NT / Debug Build  please

**Comment by Faizan2304** [comment]
@dotnet-bot test this Windows_NT / Debug Build

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Execution/ExecutionManagerTests.cs [hunk 1]
```diff
@@ -190,7 +190,7 @@ public void StartTestRunShouldAbortTheRunIfAnyExceptionComesForTheProvidedTests(
             var mockTestRunEventsHandler = new Mock<ITestRunEventsHandler>();
 
             // Call StartTestRun with faulty runsettings so that it will throw exception
-            this.executionManager.StartTestRun(new List<TestCase>(), null, @"<RunSettings><RunConfiguration><TestSessionTimeout>0</TestSessionTimeout></RunConfiguration></RunSettings>", testExecutionContext, null, mockTestRunEventsHandler.Object);
+            this.executionManager.StartTestRun(new List<TestCase>(), null, @"<RunSettings><RunConfiguration><TestSessionTimeout>-1</TestSessionTimeout></RunConfiguration></RunSettings>", testExecutionContext, null, mockTestRunEventsHandler.Object);
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Execution/ExecutionManagerTests.cs]: Should we add a unit test for positive scenario when we pass 0 ?

File: test/TestAssets/SimpleTestProject3/TestSessionTimeoutTest.cs [hunk 2]
```diff
@@ -8,19 +8,19 @@ namespace SampleUnitTestProject3
     public class TestSessionTimeoutTest
     {
         [TestMethod]
-        public void TestWhichTakeSomeTime1()
+        public void TestWhichTakeSomeTime11()
```
Comment (singhsarab) [inline test/TestAssets/SimpleTestProject3/TestSessionTimeoutTest.cs]: Why did we need this change ?

**Review by singhsarab — APPROVED** [review]
Approving with a small suggestion

**Comment by Faizan2304** [comment]
@dotnet-bot test this Windows_NT / Debug Build please

**Comment by Faizan2304** [comment]
@dotnet-bot please test this Windows_NT / Debug Build

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs [hunk 3]
```diff
@@ -114,6 +114,24 @@ public void TestSessionTimeOutTests(RunnnerInfo runnerInfo)
         }
 
         [CustomDataTestMethod]
+        [NETFullTargetFramework]
```
Comment (codito) [inline test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: I'd prefer to cover this scenario with unit tests instead of a acceptance test.

**Review by codito — APPROVED** [review]
With a suggestion to move acceptance test to unit test

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs (hunk 3 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: Sure.. Sent a new PR which is solving some other issue with TestSessionTimeout has a unittest which cover this scenario.

File: test/TestAssets/SimpleTestProject3/TestSessionTimeoutTest.cs (hunk 2 above)
Comment (Faizan2304) [reply test/TestAssets/SimpleTestProject3/TestSessionTimeoutTest.cs]: This class has 6 unittest each with 3 seconds sleep. I wanted to use three tests only so appended 1 in three test to use in TestCasefilter. But this is not usefull now as I am removing Acceptance test and adding UT for that.

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Execution/ExecutionManagerTests.cs (hunk 1 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Execution/ExecutionManagerTests.cs]: Done!!

**Pull request merged**